### DISCUSSION TITLE
Commit leaves frame around if previous magit window was dedicated

### DISCUSSION THREAD

**Discussion by Thaodan** [magit/magit#5197] [Support] [answered]
Hey,

I'm using this display-buffer function as magit-display-buffer-function to open a dedicated window
for each of them:
```emacs-lisp
(defun display-buffer-window-dedicated (buffer)
  "Like `display-buffer' but the returning window is set as dedicated."
  (when-let ((window (display-buffer buffer '(nil (inhibit-same-window . t)))))
    (set-window-dedicated-p window t)
    window))
```

For most action the window opens fine and closes however not for git commit.
When I try to cancel or  commit the frame/window that is opened for the commit stays around even after the commit buffer is closed.

How can display the buffer to commit so it is closed too?

**Comment by tarsius** [comment]
I don't know and suggest you start investigating by looking into what `kill-buffer` does, that `with-editor-finish` doesn't. Also not that emacsclient+server is involved.

I also wonder why you would need to make this window dedicated, its intended to be short-lived anyway.

**Reply by Thaodan** [reply, marked as answer]
The idea was that anything opened through the Magit buffers isn't inside the same window/frame but a different one.

If the window to open the commit editor was dedicated too it would be closed too after killing it I assume. 
If possible I would like any new file be opened with display-buffer which would open a new window.

In the end I went with this:
```emacs-lisp
(use-package with-editor
  ;; Use local Emacs instance as $EDITOR (e.g. in `git commit' or `crontab -e')
  :hook (((shell-mode eshell-mode vterm-mode term-exec) .
          with-editor-export-editor)
         ((with-editor-post-finnish with-editor-post-cancel) . delete-frame)))
```
